hack/make-rules/test-cmd-util.sh

```diff
@@ -3627,7 +3680,7 @@ run_stateful_set_tests() {
   # Pre-condition: no statefulset exists
   kube::test::get_object_assert statefulset "{{range.items}}{{$id_field}}:{{end}}" ''
   # Command: create statefulset
-  kubectl create -f hack/testdata/nginx-statefulset.yaml "${kube_flags[@]}"
+  kubectl create -f hack/testdata/rollingupdate-statefulset.yaml "${kube_flags[@]}"
```

Comment:
I initally planned to reuse `nginx-statefulset.yaml`, but found out that `updateStrategy` of the spec has to be changed. I am not sure whether it is fine to change existing YAML test files, so i created a new one.

I will remove `nginx-statefulset.yaml` if it is not used at other places.